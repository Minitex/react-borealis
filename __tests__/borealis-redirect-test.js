import React from 'react';
import { shallow } from 'enzyme';
import BorealisRoute from '../src/react-borealis-route';

describe('Borealis', () => {
  it('should render correctly', () => {
    const config = {
      pdf: {
        transcript: {
          texts: ['P.D.F ya\'ll'],
          label: 'PDF',
        },
        thumbnail: 'https://cdm16022.contentdm.oclc.org/utils/getthumbnail/collection/p16022coll35/id/0',
        config: {
          height: 800,
        },
        values: [
          {
            src: 'http://cdm16022.contentdm.oclc.org/utils/getfile/collection/p16022coll52/id/16/filename',
            thumbnail: 'http://lib-mdl-dev.oit.umn.edu/thumbnails/p16022coll52:16',
            transcript: {
              texts: ['1 Cecelia Boone Narrator Sara Ring Interviewer June 9, 2011 Minitex Oral History Project Minneapolis, Minnesota SR: Tell us how you got started working at Minitex. How did you end up in the job you are doing right now? CB: As always, there is a story. blah blah'],
              label: 'PDF',
            },
          },
          {
            src: 'http://cdm16022.contentdm.oclc.org/utils/getfile/collection/p16022coll52/id/17/filename',
            thumbnail: 'http://lib-mdl-dev.oit.umn.edu/thumbnails/p16022coll52:17',
            transcript: {
              texts: ['blah blah blah, second pdf here'],
              label: 'PDF',
            },
          },
        ],
      },
    };

    const component = shallow(<BorealisRoute config={config} basename="/" />);
    expect(component).toMatchSnapshot();
  });
});
