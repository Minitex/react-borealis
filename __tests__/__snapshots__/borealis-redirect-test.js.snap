// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Borealis should render correctly 1`] = `
<div>
  <Switch>
    <Redirect
      exact={true}
      from="/"
      push={false}
      to="/pdf/0"
    />
    <Route
      basename="image"
      exact={true}
      path="/image/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/kaltura_audio"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/kaltura_audio_playlist"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/kaltura_video_playlist"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/kaltura_video"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/audio"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/video"
      render={[Function]}
    />
    <Route
      path="/pdf/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/ppt"
      render={[Function]}
    />
    <Route
      path="/image/:id/transcript"
      render={[Function]}
    />
    <Route
      path="/kaltura_audio/transcript"
      render={[Function]}
    />
    <Route
      path="/kaltura_audio_playlist/transcript"
      render={[Function]}
    />
    <Route
      path="/kaltura_video/transcript"
      render={[Function]}
    />
    <Route
      path="/kaltura_video_playlist/transcript"
      render={[Function]}
    />
    <Route
      path="/audio/transcript"
      render={[Function]}
    />
    <Route
      path="/video/transcript"
      render={[Function]}
    />
    <Route
      path="/ppt/transcript"
      render={[Function]}
    />
  </Switch>
</div>
`;
